common/coin_support/coin_utils.c

```diff
@@ -596,4 +596,25 @@ uint16_t get_account_name(const uint32_t *path, uint16_t account_type, char *acc
     }
 
     return length;
-}
\ No newline at end of file
+}
+
+void derivation_path_array_to_string(const uint32_t *path,
+                                     const size_t path_length,
+                                     const bool harden_all,
+                                     char *output,
+                                     const size_t out_len) {
+    int offset = 0;
+    offset += snprintf(output + offset, out_len - offset, "m/");
+
+    for (int i = 0; i < path_length; i++) {
+        const bool hardened  = path[i] & 0x80000000;
+        const uint32_t value = path[i] & 0x7FFFFFFF;
```

Comment:
do test for case when out_len is less than expected.

example: `out_len = 10` when expected string is `"m/84'/0'/1000'/0/0"`